topology/probes/k8s/ingress.go

```diff
@@ -25,24 +25,50 @@ package k8s
 import (
 	"fmt"
 
+	"github.com/skydive-project/skydive/filters"
 	"github.com/skydive-project/skydive/logging"
 	"github.com/skydive-project/skydive/probe"
 	"github.com/skydive-project/skydive/topology/graph"
 
 	"k8s.io/api/extensions/v1beta1"
+	"k8s.io/apimachinery/pkg/util/intstr"
 	"k8s.io/client-go/tools/cache"
 )
 
 type ingressProbe struct {
 	defaultKubeCacheEventHandler
+	graph.DefaultGraphListener
 	*kubeCache
-	graph *graph.Graph
+	graph                *graph.Graph
+	serviceIndexer       *graph.MetadataIndexer
+	serviceIndexerByName *graph.MetadataIndexer
 }
 
 func dumpIngress(ingress *v1beta1.Ingress) string {
 	return fmt.Sprintf("ingress{Namespace: %s, Name: %s}", ingress.Namespace, ingress.Name)
 }
 
+func newServiceIndexer(g *graph.Graph) *graph.MetadataIndexer {
+	filter := filters.NewAndFilter(
+		filters.NewTermStringFilter("Manager", managerValue),
+		filters.NewTermStringFilter("Type", "service"),
+		filters.NewNotNullFilter("Namespace"),
+	)
+	m := graph.NewGraphElementFilter(filter)
+	return graph.NewMetadataIndexer(g, m, "Namespace")
+}
+
+func newServiceIndexerByName(g *graph.Graph) *graph.MetadataIndexer {
```

Comment:
serviceIndexerByName instance is never used

Reply:
removed